storage/replica.go

```diff
@@ -1755,7 +1755,12 @@ func isOnePhaseCommit(ba roachpb.BatchRequest) bool {
 	if _, hasBegin := ba.GetArg(roachpb.BeginTransaction); !hasBegin {
 		return false
 	}
-	if _, hasEnd := ba.GetArg(roachpb.EndTransaction); !hasEnd {
+	arg, hasEnd := ba.GetArg(roachpb.EndTransaction)
+	if !hasEnd {
+		return false
+	}
+	etArg := arg.(*roachpb.EndTransactionRequest)
+	if etArg.Deadline != nil && etArg.Deadline.Less(ba.Timestamp) {
```

Comment:
you could just `return etArg.Deadline == nil || !etArg.Deadline.Less(ba.Timestamp))`. The other commit path also allows `ba.Timestamp == etArg.Deadline`, correct?